cloud/shared/bin/lib/terraform.sh

```diff
@@ -18,14 +19,25 @@ function terraform::perform_apply() {
     terraform \
       -chdir="${TERRAFORM_TEMPLATE_DIR}" \
       init \
+      -input=false \
       -upgrade \
       -backend-config="${BACKEND_VARS_FILENAME}"
   fi
   
-  terraform \
-    -chdir="${TERRAFORM_TEMPLATE_DIR}" \
-    apply \
-    -var-file="${TF_VAR_FILENAME}"
+  if azure::is_service_principal; then
+    terraform \
+        -chdir="${TERRAFORM_TEMPLATE_DIR}" \
```

Comment:
[nit] Might be worth putting this into a variable, so we know to edit both commands together (and just add the extra flag on the end)